orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/QuickPatchStage.groovy

```diff
@@ -53,16 +56,21 @@ class QuickPatchStage implements StageDefinitionBuilder {
 
   public static final String PIPELINE_CONFIG_TYPE = "quickPatch"
 
-  private static INSTANCE_VERSION_SLEEP = 10000
-
   @Override
-  def void taskGraph(Stage stage, TaskNode.Builder builder) {
+  void taskGraph(Stage stage, TaskNode.Builder builder) {
     builder.withTask("resolveQuipVersion", ResolveQuipVersionTask)
   }
 
   @Override
-  def List<Stage> aroundStages(Stage stage) {
-    def stages = []
+  void beforeStages(@Nonnull Stage parent, @Nonnull StageGraphBuilder graph) {
+    // mark as SUCCEEDED otherwise a stage w/o child tasks will remain in NOT_STARTED
+    parent.status = ExecutionStatus.SUCCEEDED
+  }
+
+  @Override
+  void afterStages(@Nonnull Stage stage, @Nonnull StageGraphBuilder graph) {
+    ArrayList<Stage> stages = new ArrayList<>()
+    HashMap<String, Object> nextStageContext = new HashMap<>()
```

Comment:
Generally speaking, you shouldn't cast the variable with the implementation class, preferring the most common interface that you require. There are exceptions, but I don't believe this is one of them. There's a few spots that this is done in this PR that could be tidied a bit.

Reply:
thanks!